ttnn/cpp/ttnn/tensor/tensor.hpp

```diff
@@ -344,20 +345,13 @@ void memcpy(
 void memcpy(Tensor &dst, const void *src, const std::optional<std::size_t> transfer_size = std::nullopt);
 void memcpy(Tensor &dst, const Tensor &src, const std::optional<std::size_t> transfer_size = std::nullopt);
 
-Tensor allocate_tensor_on_device(
-    const ttnn::Shape &shape,
-    DataType data_type,
-    Layout layout,
-    Device *device,
-    const MemoryConfig &memory_config = {.memory_layout = tt::tt_metal::TensorMemoryLayout::INTERLEAVED},
-    const std::optional<Tile>& tile = std::nullopt);
-Tensor allocate_tensor_on_device(
+Tensor allocate_tensor_on_workers(
     const ttnn::Shape &shape,
     DataType data_type,
     Layout layout,
-    distributed::MeshDevice *mesh_device,
+    const std::vector<Device *> &workers,
```

Comment:
Wouldn't this interface be cleaner as single object handle of `ttnn::SimpleDevice` to simplify caller-side and more toward the eventual goal of having single handle?

Reply:
Yeah, I'm looking into refactoring this in follow up PRs. Right now the main blocker is that sometimes we only have a `Tensor` and we need to get the backing devices. So we either need to introduce `SimpleDevice` into `Tensor`, or do a more substantial refactoring to decouple allocation functions from `Tensor` (e.g. supply `SimpleDevice` externally, where it is needed).

Reply:
I didn't see anything in this changelist that would've blocked us from doing something like:

```
Tensor allocate_tensor_on_device(
    const ttnn::Shape& shape,
    DataType data_type,
    Layout layout,
    SimpleDevice device,
    const MemoryConfig& memory_config,
    const std::optional<Tile>& tile) 
{
...
simple_device.get_devices()
}

```

Reply:
But I do agree that we'd need to introduce `SimpleDevice` into Tensor for us to propagate this more fully. I'm interested in seeing how Patrick's draft PR will look like. Until then, if it helps unblock @dmakoviichuk-tt as a stop-gap, it's fine.

Reply:
In this PR, there are a couple of places where we need to get a vector of workers from a given tensor; for example this snippet in `EmptyLike`:
```
struct EmptyLike {
    static ttnn::Tensor invoke(
        const ttnn::Tensor& tensor,
        ...
        detail::OptionalSimpleDevice device_arg = std::nullopt,
        ...) {
        const std::vector<Device*>& devices =
            device_arg.has_value() ? device_arg->get_devices() : tensor.get_workers(/*blocking=*/true);
        ...
        return allocate_tensor_on_workers(tensor.get_shape(), dtype_value, layout_value, devices, mem_cfg);
    }
};
```

+1 to the longer term plan.

Reply:
Makes sense.. We can clean this up once some of the refactor happens.